lib/util/utility.dart

```diff
@@ -0,0 +1,55 @@
+import 'dart:convert';
+
+import 'package:Solon/models/comment.dart';
+import 'package:Solon/models/event.dart';
+import 'package:Solon/models/forum_post.dart';
+import 'package:Solon/models/model.dart';
+import 'package:Solon/models/proposal.dart';
+import 'package:Solon/util/user_util.dart';
+import 'package:flutter/material.dart';
+import 'package:http/http.dart' as http;
+
+class Utility {
+  static Stream<List<T>> getList<T extends Model>({
+    @required Function function,
+    @required String body,
+    @required Type type,
+    String query,
+    int fid,
+  }) async* {
+    http.Response response = (fid != null) ? await function(fid: fid) : await function(query: query);
+    String prefLangCode = await UserUtil.getPrefLangCode();
+    List temp;
+    List<Model> collection;
+    if (response.statusCode == 200) {
+      temp = json.decode(response.body)[body];
+      switch (type) {
+        case Proposal:
+          collection = temp
+              .map((json) =>
+                  Proposal.fromJson(map: json, prefLangCode: prefLangCode))
+              .toList();
+          break;
+        case ForumPost:
+          collection = temp
+              .map((json) =>
+                  ForumPost.fromJson(map: json, prefLangCode: prefLangCode))
+              .toList();
+          break;
+        case Event:
+          collection = temp
+              .map((json) =>
+                  Event.fromJson(map: json, prefLangCode: prefLangCode))
+              .toList();
+          break;
+        case Comment:
+          collection = temp
+              .map((json) =>
+                  Comment.fromJson(map: json, prefLangCode: prefLangCode))
+              .toList();
+          break;
```

Comment:
please tell me there's a way to abstract this 
look into casting generic types in super constructors?

Reply:
maybe use an enum ?

Reply:
collection = temp
              .map((json) =>
                  T.fromJson(map: json, prefLangCode: prefLangCode))
              .toList();

Reply:
would that work or no; assuming/asserting that T has a fromJson function

Reply:
tried this several times but T isnt assumed to have a fromJson function even if it extends from Model(?)

Reply:
you may need to move fromJson into model and then override it in proposal/event/forum

Reply:
bc if you do T extends Model, and model doesnt have fromJson, then that might be why T can't assume to have a fromJson function

Reply:
you override fromJson in proposal/event/forum bc those 3 fxns are quite diff from one another and idt that should be abstracted further

Reply:
You can take a swing at it but I think I've taken a swing at every conceivable combination of abstract classes, interfaces, implements, etc. to no avail.

Reply:
i think repeating some code is fine here then

Reply:
ill just backlog it to keep it in mind in the future when we find a better way